### PR TITLE
The great rewrite

### PR DESCRIPTION
**Pull request by rcoedo**

### CODE REVIEW

File: src/registry.js [hunk 1]
```diff
@@ -1,64 +1,187 @@
...
+  // force a render for all outlets connected with a given inlet
+  updateConduits = inlet => this.findConduitsByInlet(inlet).forEach(conduit => conduit.update());
+
+  // build children using all inlets for the given label
+  mergeChildrenForLabel = label =>
+    createFragment(
+      this.findInletsByLabel(label)
+        .sort((i1, i2) => i1.index - i2.index)
+        .map(inlet => [inlet.getId(), inlet.getChildren()])
+        .reduce((acc, [id, children]) => ({ ...acc, [id]: children }), {}),
```
Comment (asis) [inline src/registry.js:181]: Do you need to sort the inlets? I mean, you are building a new object, which is unordered by definition (IIRC).

Comment (rcoedo) [reply src/registry.js:181]: Object enumeration order is not guaranteed by the spec, however most vm's implement it anyway. createFragment uses this "feature" (😅 ) to determine the order of the children.

File: src/conduit.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  connect = () => {
+    this.inlet.onConnect(this);
+    this.outlet.onConnect(this);
+  };
+
+  getId = () => `${this.inlet.getId()}-${this.outlet.getId()}`;
+  getInlet = () => this.inlet;
+  getOutlet = () => this.outlet;
+  update = () => this.outlet.forceRender();
+  simplify = () => ({ inlet: this.inlet, outlet: this.outlet });
```
Comment (davidbarral) [inline src/conduit.js]: IMHO simplify is a very obscure name ^_^U. Here and in all the places where it is used.

Comment (rcoedo) [reply src/conduit.js]: It is, any suggestion? 😉

Comment (davidbarral) [reply src/conduit.js]: 🙃
¿components? ¿members? ¿getInletAndOultet? ¿conduitEnds? ¿conduitPoles? 
Who knows!

Comment (rcoedo) [reply src/conduit.js]: 👍

Comment (davidbarral) [reply src/conduit.js]: Much more sexy now ;)

**Pull request merged**